docker/r10k/Dockerfile

```diff
@@ -49,10 +52,11 @@ COPY --from=build /workspace/r10k.gem /
 SHELL ["/bin/ash", "-eo", "pipefail", "-c"]
 # ignore apk and gem pinning
 # hadolint ignore=DL3018,DL3028
-RUN chmod a+x /adduser.sh /docker-entrypoint.sh && \
+RUN chmod -R a+x /adduser.sh /docker-entrypoint.sh /docker-entrypoint.d && \
 # Add a puppet user to run r10k as for consistency with puppetserver
     /adduser.sh && \
-    chown -R puppet: /docker-entrypoint.d /docker-entrypoint.sh && \
+    chown -R "${PP_USER}:" /docker-entrypoint.d /docker-entrypoint.sh && \
```

Comment:
Note that permissions here are being modified at *build time*... but then later consumed at runtime in the entrypoint script. So if a user ever set `PP_USER` to something other than `puppet`, you wouldn't get the behavior you're aiming for.